### DISCUSSION TITLE
Clarification on DeviceGroups and UDP Usage

### DISCUSSION THREAD

**Discussion by laguill** [arendst/Tasmota#22610] [Q&A]
Hello,

I have some questions regarding the implementation and usage of DeviceGroups:

- What underlying technology is used by DeviceGroups?
- Is the use of UDP for DeviceGroups considered secure? Are there any specific precautions to be aware of?
- Is it possible to send a UDP request to a DeviceGroup from a terminal? If so, could you provide an example?

Thank you for your assistance!

**Comment by sfromis** [comment]
Yeah, the "underlying technology" **is** UDP. Since UDP as such cannot be "secure", and no security protocol is implemented on top (assuming that mild "security by obscurity" does not count), the usual theme with perimeter security also applies here. This is only an additional reason why you should just not leave your network open to non-trusted actors. You'd get better security with MQTT, even without MQTT-TLS.

It would certainly be "possible" to make similar UDP requests from suitable tools, but the internal protocol is not meant as an API; hence no documentation nor reasonable expectations of stability and transparent behavior. Of course, it would not be that hard to explore how it works, and it is likely not getting much changed, but that does not make it an API.

To have access from outside of Tasmota, the simplest would be to have two Tasmota devices grouped, with one having "virtual" relays (or whatever) doing nothing, and the other device will then receive the operations performed on the virtual outputs.

**Comment by laguill** [comment]
Thank you for detailed reply.

Is it possible to disable the UDP function in DeviceGroups but continue using it by transmitting requests through an MQTT server instead?

Here is my current setup:

I use DeviceGroups to synchronize multiple smart bulbs with a single switch.
I have configured rules on the switch to:
Control the bulbs' power state.
Change the bulbs' colors depending on the time of day.
If I switch to using MQTT instead of UDP for the same functionality:

Can I achieve the same result?
Will I need to rewrite all my existing configurations and rules?
Thank you!

Here is a Rule I am currently using maybe not fully optimised but

```haskell
Rule1 
  on Power1#State=1 do backlog Delay 15; event checkdate=%timestamp% endon 
  on event#checkdate$|-10-31T do Backlog Palette 4 9; Speed 15; Dimmer 100; Scheme 4; Power 1;  Break
  on event#checkdate$|-12-08T do Backlog Palette BF79000000 0000000F57 BF7F000000; Speed 15; Fade 1; Dimmer 75; Scheme 4; Power 1 Break
  on event#checkdate$|-12-24T do Backlog Palette 1 3 6 8; Speed 15; Dimmer 100; Scheme 4; Power 1; Break
  ON event#checkDate DO Backlog Palette 0; Speed 1; Scheme 0; Dimmer 100; CT 500; ENDON
```

**Reply by sfromis** [reply]
No, device groups are only via UDP. And while you can send any command via MQTT, creating rules to pick up on changes and transmit them is tedious, as rules can only pick up one value at a time, excluding dynamically changing values.

With the rule set you posted, you could extend it to issue the same command locally and via MQTT.
